base-template.sh

```diff
@@ -15,7 +15,8 @@
 #   0 when successfully finished, 1 otherwise
 #####################################################
 process_git_hook() {
-    set_main_variables "$@"
+    set_main_variables "$@" && shift && shift
```

Comment:
This looks a bit unclear to me. Would it be better to shift at the place where `$1` and `$2` are interpreted? Or set them where the `shift` is?

Also, a comment about the shift might be helpful (e.g. shift off extra arguments added by symlink, leaving only the original git-provided arguments in `$@`).